src/components/Booklist.js

```diff
@@ -1,14 +1,25 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import Books from './Books';
 import FormInput from './FormInput';
+import { fetchBooks } from '../redux/books/books';
 
 const Booklist = () => {
   const Booklist = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchBooks());
+  });
```

Comment:
- [ ] On the first loading of your page the component continues to re-render. Kindly fix that. The component should be stable on first refresh and should not continue to re-render.

  You can fix this by adding an empty array, [], as a dependency for this `useEffect`.

  ```
  useEffect(() => {
    dispatch(fetchBooks());
  }, []);
  ```